src/sidebar/components/ShareDialog/test/ShareDialog-test.js

```diff
@@ -119,24 +115,30 @@ describe('ShareDialog', () => {
     assert.equal(activeTabPanel(wrapper).props().id, 'share-panel');
   });
 
-  describe('a11y', () => {
-    beforeEach(() => {
-      enableFeature('export_annotations');
-      enableFeature('import_annotations');
+  it('renders empty if no tabs should be displayed', () => {
+    const wrapper = createComponent({
+      shareTab: false,
+      exportTab: false,
+      importTab: false,
     });
 
+    assert.isFalse(wrapper.exists('TabHeader'));
+    assert.isFalse(wrapper.exists('ShareAnnotations'));
+  });
+
+  describe('a11y', () => {
     it(
       'should pass a11y checks',
       checkAccessibility({
         content: () =>
           // ShareDialog renders a Fragment as its top-level component when
-          // `export_annotations` feature is enabled.
+          // it has import and/or export tabs.
           // Wrapping it in a `div` ensures `checkAccessibility` internal logic
           // does not discard all the Fragment children but the first one.
           // See https://github.com/hypothesis/client/issues/5671
           mount(
             <div>
-              <ShareDialog />
+              <ShareDialog shareTab exportTab importTab />
```

Comment:
Eventually this API would get a bit clunky if you added more tabs. In that case you could revert back to using an array of tab IDs, but inside the component do something like `const shareTab = tabs.includes('share')` for each tab.